Jenkinsfile

```diff
@@ -158,27 +158,57 @@ pipeline {
                 )
             }
         }
-        stage("Create a 'dev' and 'latest' tag from 'master'") {
+        stage("Create a 'latest' tag from 'master'") {
             when {
                 branch 'master'
             }
-            environment {
-                // qe-pipeline-library step
-                DOCKER_DEV_TAG = getDevTag("${env.CLONED_REPOSITORY_PATH}")
-            }
             steps{
                 // qe-pipeline-library step
                 tagRemoteContainerImage(
                     credentialsId: "${env.CREDENTIALS_ID}",
                     sourceImage: "${env.OPERATOR_CONTAINER_IMAGE_NAME}",
-                    targetImage: "quay.io/aerogear/${env.OPERATOR_NAME}:${env.DOCKER_DEV_TAG}",
+                    targetImage: "${env.OPERATOR_CONTAINER_IMAGE_NAME_latest}",
                     deleteOriginalImage: false
                 )
+            }
+        }
+        // rebuild and push the image so that it points to master tag of operand (for nightly testing purposes)
+        stage("Rebuild and push 'dev' image for nightly testing") {
+            when {
+                branch 'master'
+            }
+            environment {
+                // qe-pipeline-library step
+                DOCKER_DEV_TAG = getDevTag("${env.CLONED_REPOSITORY_PATH}")
+            }
+            steps{
+                dir("${env.CLONED_REPOSITORY_PATH}") {
+                    // remove original build
+                    sh "rm -rf build/_output"
+
+                    // replace specific operand version with 'master'
+                    sh 'sed -i -E \'s/"MDC_IMAGE_STREAM_TAG", "[a-zA-Z0-9.+-]*"/"MDC_IMAGE_STREAM_TAG", "master"/\' pkg/config/config.go'
+                    sh 'sed -i -E \'s/quay.io\\/aerogear\\/mobile-developer-console:[a-zA-Z0-9.+-]*/quay.io\\/aerogear\\/mobile-developer-console:master/\' pkg/config/config.go'
+
+                    // compile
+                    sh "make code/compile"
+
+                    // build image with full dev tag
+                    // qe-pipeline-library step
+                    dockerBuildAndPush(
+                        credentialsId: "${env.CREDENTIALS_ID}",
+                        containerRegistryServerName: "quay.io",
+                        containerImageName: "quay.io/aerogear/${env.OPERATOR_NAME}:${env.DOCKER_DEV_TAG}",
```

Comment:
You could conceivably put these testing images into a different repo (even different org) if you wanted to. No real reason, other than maybe not wanting to pollute that repo with lots of testing images.

Reply:
Another alternative is to just build these custom images as part of the testing pipeline altogether...this would reduce the chance of us building multiple of them throughout the day, and your testing only consuming one of them.

Reply:
Separate testing repo/org should be possible, just thinking that for every PR there are at least 2 images now published - image for PR, then one when PR is merged. This change would add only one more image for every merged PR.

Building the images in the testing pipeline would also be possible, only drawback I can see is that if later we move to some other ci for building the images, we may risk that build environments can slightly differ, and we may be testing something bit different than what will go to release later.